components/postmark/sources/new-inbound-email-received/new-inbound-email-received.mjs

```diff
@@ -0,0 +1,36 @@
+import postmark from "../../postmark.app.mjs";
+
+export default {
+  key: "postmark-new-inbound-email-received",
+  name: "New inbound email received",
+  description:
+    "Emit new event when an email is received by the Postmark server",
+  version: "0.0.1",
+  type: "source",
+  props: {
+    postmark,
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+  },
+  hooks: {
+    async activate() {
+      return this.postmark.setServerInfo({
+        InboundHookUrl: this.http.endpoint,
+      });
+    },
+    async deactivate() {
+      return this.postmark.setServerInfo({
+        InboundHookUrl: "",
+      });
+    },
+  },
```

Comment:
You can move all common things of sources to a common file.

Example: https://github.com/PipedreamHQ/pipedream/blob/master/components/asana/sources/common/common.mjs

Reply:
Good examples! I created a `common` module for the sources and managed to share a good deal of their inner workings via this file.